homeassistant/components/nuki/sensor.py

```diff
@@ -0,0 +1,68 @@
+"""Battery sensor for the Nuki Lock."""
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import NukiEntity
+from .const import ATTR_NUKI_ID, DATA_COORDINATOR, DATA_LOCKS, DOMAIN as NUKI_DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Nuki lock sensor."""
+    data = hass.data[NUKI_DOMAIN][entry.entry_id]
+    coordinator = data[DATA_COORDINATOR]
+
+    entities = []
+
+    for lock in data[DATA_LOCKS]:
+        entities.append(NukiBatterySensor(coordinator, lock))
+
+    async_add_entities(entities)
```

Comment:
This could be a comprehension
```suggestion
    async_add_entities(
        [
            entities.append(NukiBatterySensor(coordinator, lock))
            for lock in data[DATA_LOCKS]
        ]
    )
```